### PR TITLE
Log alive hpx threads on exit

### PR DESCRIPTION
**Pull request by Pansysk75**
Adds logging when an HPX program is terminated suddenly (ie by a termination signal, addresses issue #6329).
I'll leave it as draft until I make up my mind on whether I think it is actually a good idea

### CODE REVIEW

File: libs/core/runtime_local/src/runtime_local.cpp [hunk 1]
```diff
@@ -57,6 +57,27 @@
...
+                    LTM_(debug).format(
+                        "Logging all runtime threads: pool({}), scheduler({}),"
+                        "thread({}), description({}), state({})",
+                        sched->get_parent_pool(), sched, id,
+                        td->get_description(), td->get_state().state());
+                    return true;
+                });
+        }
+    }
+}};    // namespace hpx::detail
```
Comment (hkaiser) [inline libs/core/runtime_local/src/runtime_local.cpp:79]: Should we wrap this in a `#if defined(HPX_HAVE_LOGGING)` as well?

Comment (hkaiser) [reply libs/core/runtime_local/src/runtime_local.cpp:79]: I'd also suggest to wrap all of the body into a `try {} catch (...) {}` as this is being invoked in a signal handler and we can't be sure that all of the runtime is in a consistent and valid state at this point any more.

**Review by hkaiser — CHANGES_REQUESTED** [review]

File: libs/core/runtime_local/src/runtime_local.cpp (hunk 1 above)
Comment (Pansysk75) [reply libs/core/runtime_local/src/runtime_local.cpp:87]: Thanks!

**Review by hkaiser — APPROVED** [review]
LGTM, thanks!

**Comment by hkaiser** [comment]
bors merge

**Pull request merged**